src/backend/joanie/tests/payment/test_backend_payplug.py

```diff
@@ -889,3 +890,222 @@ def test_payment_backend_payplug_abort_payment_request_failed(
                 "The server gave the following response: `Abort this payment is forbidden.`."
             ),
         )
+
+    @mock.patch.object(BasePaymentBackend, "_send_mail_refused_debit")
+    @mock.patch.object(payplug.notifications, "treat")
+    def test_payment_backend_payplug_payment_failure_with_installment_should_trigger_email_method(
+        self, mock_treat, mock_send_mail_refused_debit
+    ):
+        """
+        When backend receives a payment notification which failed, the generic method
+        `_do_on_payment_failure` should be called and also call the method that is responsible
+        to send an email to the user.
+        """
+        payment_id = "pay_failure"
+        user = UserFactory(
+            first_name="John",
+            last_name="Doe",
+            language="en-us",
+            email="john.doe@acme.org",
+        )
+        product = ProductFactory(price=D("999.99"), title="Product 1")
+        order = OrderFactory(
```

Comment:
For the lyra backend tests you are using the `OrderGeneratorFactory` factory. IMO you should use it here for the new tests.